### PR TITLE
Feature/fiat service

### PR DESCRIPTION
**Pull request by anskaal**
Depends on https://github.com/toshiapp/toshi-headless-client/pull/21

### CODE REVIEW

File: src/lib/Fiat.js [hunk 1]
```diff
@@ -1,66 +1,93 @@
...
-        return fiat / rates[code];
-      };
-      fn.fromEth = function(fiat) {
-        return (fiat * rates[code]).toFixed(2);
-      };
-      helpers[code] = fn;
-    })(code,rate)
-  }
+function _getUrl(path) {
+    return _base_url + path;
```
Comment (tristan) [inline src/lib/Fiat.js]: having this function be global along with the _base_url variable is pointless, it's not needed outside of the `FiatService`  class so just put both in there (like the IdService does)

File: src/lib/Fiat.js [hunk 2]
```diff
@@ -1,66 +1,93 @@
...
+        };
 
-getRates()
-  .then((helper) => {
-    Logger.info("Fiat: Rates initialized successfully");
-  }).catch((err) => {
-    Logger.error(err);
-  });
+        Logger.info("Fiat: Fetching rates");
+        return fetch(_getUrl('/v1/rates/'))
```
Comment (tristan) [inline src/lib/Fiat.js]: This is actually wrong! it needs something at the end! e.g. `/v1/rates/ETH`

File: src/lib/Fiat.js [hunk 3]
```diff
@@ -1,66 +1,93 @@
...
+        }
+        else {
+            Logger.debug("Fiat: Using cached rates");
+            return Promise.resolve(helpers)
+        }
+    }
+}
 
-module.exports = { fetch: fetch, rates: rates };
+module.exports = new FiatService();
```
Comment (tristan) [inline src/lib/Fiat.js]: might be good to add in a getter to the class to fulfil the `rates` export, incase someone is using it
```
class FiatService {
  ...
  get rates() {
    return rates;
  }
  ...
}
```

File: src/lib/Fiat.js [hunk 4]
```diff
@@ -1,66 +1,93 @@
...
-function fetch(limit=CACHE_AGE_LIMIT) {
-  let now = new Date().getTime()
-  if (now - cachedAt > limit) {
-    return getRates()
-  } else {
-    Logger.debug("Fiat: Using cached rates")
-    return Promise.resolve(helpers)
-  }
-}
+class FiatService{
```
Comment (tristan) [inline src/lib/Fiat.js]: 👀

File: src/lib/Fiat.js (hunk 3 above)
Comment (anskaal) [reply src/lib/Fiat.js]: @tristan `rates` is a property and you don't need to access it dynamically, so what is the point?

File: src/lib/Fiat.js (hunk 1 above)
Comment (anskaal) [reply src/lib/Fiat.js]: It wouldn't be global from a node-module unless you explicitly expose it, but I agree it's pointless when I think about it. The developer has access to the source code anyway, so I guess there's no point in trying to make it "private".

File: src/lib/Fiat.js (hunk 4 above)
Comment (anskaal) [reply src/lib/Fiat.js]: :see_no_evil:

File: src/lib/Fiat.js (hunk 3 above)
Comment (tristan) [reply src/lib/Fiat.js]: no you're right, i just noticed this myself!

**Comment by anskaal** [comment]
Added some whitespace for you as well @tristan

File: src/lib/Fiat.js [hunk 5]
```diff
@@ -1,66 +1,85 @@
...
-    Logger.error(err);
-  });
+        Logger.info("Fiat: Fetching rates");
+        return fetch(this._getUrl('/v1/rates/ETH'))
+            .then((body) => {
+                cachedAt = new Date().getTime();
+
+                let obj = JSON.parse(body);
+                for (let k in obj.rates) {
+                    if(obj.rates.hasOwnProperty(k)){
```
Comment (tristan) [inline src/lib/Fiat.js]: 👀

**Pull request merged**